codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocket13FrameDecoder.java

```diff
@@ -139,28 +136,28 @@ public WebSocket13FrameDecoder(boolean expectMaskedFrames, boolean allowExtensio
      *            Frames decoder configuration.
      */
     public WebSocket13FrameDecoder(WebSocketDecoderConfig decoderConfig) {
-        this.config = Objects.requireNonNull(decoderConfig, "decoderConfig");
+        config = Objects.requireNonNull(decoderConfig, "decoderConfig");
     }
 
-    private static int toFrameLength(long l) {
-        if (l > Integer.MAX_VALUE) {
-            throw new TooLongFrameException("Length:" + l);
+    private static int toFrameLength(long length) {
+        if (length > Integer.MAX_VALUE) {
+            throw new TooLongFrameException("Length:" + length);
         } else {
-            return (int) l;
+            return (int) length;
         }
     }
 
     @Override
-    protected void decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception {
+    protected void decode(ChannelHandlerContext ctx, Buffer in) throws Exception {
         // Discard all data received if closing handshake was received before.
         if (receivedClosingHandshake) {
-            in.skipBytes(actualReadableBytes());
+            in.skipReadable(actualReadableBytes());
             return;
         }
 
         switch (state) {
-        case READING_FIRST:
-            if (!in.isReadable()) {
+        case READING_FIRST: {
+            if (in.readableBytes() <= 0) {
```

Comment:
I think this never can be negative or I am wrong ?
```suggestion
            if (in.readableBytes() == 0) {
```

Reply:
Yeah, I think so.